src/scripts/main.js

```diff
@@ -1,3 +1,163 @@
 'use strict';
 
-// write your code here
+let matrix = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+const message = [
+  document.querySelector('.message-start'),
+  document.querySelector('.message-win'),
+  document.querySelector('.message-lose'),
+];
+const buttonStart = document.querySelector('.start');
+const gameFieldCells = document.querySelectorAll('.field-cell');
+const infoScore = document.querySelector('.game-score');
+let gameScore = +(infoScore.textContent);
+const ROW_LENGTH = 4;
+const PROBILITY_NUMBER = 0.1;
+
+addRandomCell();
+addRandomCell();
+
+buttonStart.addEventListener('click', () => {
+  if (buttonStart.classList.contains('start')) {
+    goStart();
+  } else {
+    goEnd();
+  }
+});
+
+function goStart() {
+  buttonStart.classList.remove('start');
+  buttonStart.classList.add('restart');
+  buttonStart.textContent = 'restart';
+
+  message[0].classList.add('hidden');
+  addRandomCell();
+};
+
+function goEnd() {
+  buttonStart.classList.remove('restart');
+  buttonStart.classList.add('start');
+  buttonStart.textContent = 'start';
+
+  message[0].classList.remove('hidden');
+
+  matrix = matrix.map(row => row.map(() => 0));
+
+  gameFieldCells.forEach(cell => {
+    cell.textContent = '';
+    cell.className = 'field-cell';
+  });
+  gameScore = 0;
+  addRandomCell();
+  addRandomCell();
+};
```

Comment:
You're adding random cells after resetting the game, but you haven't considered the case when a player may lose right after the game restarts. You should check if there are valid moves before adding new random cells.